File: src/App/Models/ServiceCategories.cs
namespace AppointWeb.Api.Models;

public static class ServiceCategories
{
    public const string HealthcareAndDental = "Healthcare & Dental";
    public const string SportsAndFitness = "Sports & Fitness";
    public const string BeautyAndWellness = "Beauty & Wellness";
    public const string ClassesAndCoaching = "Classes & Coaching";
    public const string Other = "Other";

    public static readonly string[] All =
    [
        HealthcareAndDental,
        SportsAndFitness,
        BeautyAndWellness,
        ClassesAndCoaching,
        Other,
    ];

    public static bool IsValid(string? category) =>
        !string.IsNullOrWhiteSpace(category) &&
        All.Contains(category.Trim(), StringComparer.Ordinal);
}

File: src/App/ClientApp/src/constants/serviceCategories.ts
export const SERVICE_CATEGORIES = [
  "Healthcare & Dental",
  "Sports & Fitness",
  "Beauty & Wellness",
  "Classes & Coaching",
  "Other",
] as const;

export type ServiceCategory = (typeof SERVICE_CATEGORIES)[number];

export const OTHER_CATEGORY: ServiceCategory = "Other";

export function isServiceCategory(value: string): value is ServiceCategory {
  return SERVICE_CATEGORIES.includes(value as ServiceCategory);
}
